### PR TITLE
Support of graphqlite 7 and symfony 7

### PR DESCRIPTION
**Pull request by fogrye**
As symfony 7 completely drops support of doctrine annotations I think it's a great timing to drop them and fix this project to work with attributes.

### CODE REVIEW

File: src/Annotations/Assertion.php [hunk 1]
```diff
@@ -21,26 +22,34 @@
...
     /** @var Constraint[] */
-    private $constraint;
+    private array $constraint;
 
-    /** @param array<string, mixed> $values */
-    public function __construct(array $values)
-    {
-        if (! isset($values['for'])) {
-            throw new BadMethodCallException('The @Assert annotation must be passed a target. For instance: "@Assert(for="$email", constraint=@Email)"');
+    /**
```
Comment (nguyenk) [inline src/Annotations/Assertion.php:32]: This looks like a breaking change, no ? I mean people upgrading might get errors if they are using Annotations.

**Comment by nguyenk** [comment]
Hello @fogrye , thanks a mile for your PR. I'm afraid we might have a breaking change while supporting both version 6 and 7 and introducing attributes. If someone using version 6 with Annotations, they will get errors using `src/Annotations/Assertion.php`  as an Annotation.

What do you think ?

**Comment by fogrye** [comment]
@nguyenk doctrine annotations are deprecated for a long time and were finally removed in v7, technically somebody can have an issue in symfony 6 with attributes but we have to make it that way

File: src/Annotations/Assertion.php (hunk 1 above)
Comment (fogrye) [reply src/Annotations/Assertion.php:32]: yes that's BC to support symfony 7

**Comment by SCIF** [comment]
@nguyenk , attributes is the way. `rector` can help with migration. Why not just release v7.0 of the bridge and drop symfony v6 support?

**Comment by fogrye** [comment]
> drop symfony v6 support

Symfony v6 is in LTS, I would not drop that.

**Comment by SCIF** [comment]
> > drop symfony v6 support
> 
> Symfony v6 is in LTS, I would not drop that.

I mean to drop the support of symfony v6 in bridge v7. But bridge v6 can still be maintained/supported.

**Comment by fogrye** [comment]
@nguyenk so I prepared PR to be pushed as v7 because of BC.

**Review by SCIF — APPROVED** [review]

**Comment by fogrye** [comment]
Hi @nguyenk I'm also [participating in removal of annotations support for graphqlite](https://github.com/thecodingmachine/graphqlite/issues/675) so I think it's completely ok to drop them in this project. Waiting patiently for your approval.

**Comment by fogrye** [comment]
@nguyenk I see you're not so active lately, I can help you with maintenance of this lib so it would not become a blocker for graphqlite-bundle

**Comment by nguyenk** [comment]
Hello folks ! Thank you both for you contribution @SCIF and @fogrye. I'm currently off, and was previously very busy. I did see your proposal to contribute more actively. Let's discuss this further next time I become a blocker again :).

v7.0 has just been released.

@fogrye I guess you would like to continue with this one ? https://github.com/thecodingmachine/graphqlite-bundle/pull/203

Cheers

**Pull request merged**